Source/OCMock/OCMockObject.m

```diff
@@ -261,50 +284,75 @@ - (void)forwardInvocation:(NSInvocation *)anInvocation
     }
     @catch(NSException *e)
     {
-        [exceptions addObject:e];
+        @synchronized(exceptions)
+        {
+            [exceptions addObject:e];
+        }
         [e raise];
     }
 }
 
 - (BOOL)handleInvocation:(NSInvocation *)anInvocation
 {
-    [invocations addObject:anInvocation];
-
+    @synchronized(invocations)
+    {
+        // When the method has a char* argument we do not retain the arguments. This makes it possible
+        // to match char* args literally and with anyPointer. Not retaining the argument means that
+        // in these cases tests that use their own autorelease pools may fail unexpectedly.
+        if(![anInvocation hasCharPointerArgument])
+            [anInvocation retainArguments];
+        [invocations addObject:anInvocation];
+    }
+    
```

Comment:
Another option would be to clear the target before calling -retainArguments, and when actually needing to *use* the invocation, make a copy of the invocation and call invokeWithTarget: on that.  If we can assume the target is always "self".

I did something like this in https://github.com/Collect3/CTAppearance/pull/4 .

And really, arguments need to be retained if any of them are objects, not just if one of them happens to be a char * -- that seems a little odd.

Requiring calls to -stopMocking when there are untold thousands of existing lines of test case code out there written over many years which would suddenly turn into memory leaks without -stopMocking is probably not the best idea.